### PR TITLE
Authentication for user API key

### PR DESCRIPTION
**Pull request by makistsantekidis**
- Added function in core package to check user api key
- Added django view to check user key in header

### CODE REVIEW

**Comment by themiszamani** [comment]
a) document your code
b) add api call documentation based on the template
c) for error response as a top level i think we should use errors instead of unauthorized (maybe with an array of error objects). What do you think?
d) for success response please keep only the necessary fields and use as a top level data or something more descriptive. 
e) do we need the Success \n and Unauthorized Access \n message?

**Comment by makistsantekidis** [comment]
please review @themiszamani @gouzouni625

File: webapp/api-doc/Authentication.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+| **URL**         | /backend/authentication         |
+| **HTTP Method** | GET                             |
+| **Security**    | Basic Authentication            |
+
+
+### Headers 
+
+Type | Description | Required | Default value | Example value |
+------|-------------|----------|---------------|---------------|
+token | ....... | `Yes` |None| |
```
Comment (themiszamani) [inline webapp/api-doc/Authentication.md]: correct the row and add description .

File: webapp/api-doc/Authentication.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+### Response messages
+
+The main response messages are: 
+
+- HTTP/1.1 200 OK : (Success)
+- HTTP/1.1 401 UNAUTHORIZED : (Fail)
+
+
```
Comment (themiszamani) [inline webapp/api-doc/Authentication.md:71]: I think we should add the error response too

**Pull request merged**